lib/src/clojure_lsp/internal_api.clj

```diff
@@ -351,24 +352,63 @@
                  (assoc a uri diagnostics))
                {})))
 
-(defn ^:private diagnostics* [{{:keys [format]} :output :as options} {:keys [db*] :as components}]
-  (setup-api! components)
-  (setup-project-and-clojure-only-deps-analysis! options components)
-  (cli-println options "Finding diagnostics...")
-  (let [db @db*
-        diags-by-uri (diagnostics-by-uri db options)
-        diags (mapcat val diags-by-uri)
-        errors? (some (comp #(= 1 %) :severity) diags)
-        warnings? (some (comp #(= 2 %) :severity) diags)]
-    (if (seq diags-by-uri)
-      {:result-code (cond errors? 3 warnings? 2 :else 0)
-       :message-fn (fn []
-                     (case format
-                       :edn (with-out-str (pr diags-by-uri))
-                       :json (json/generate-string diags-by-uri)
-                       (string/join "\n" (diagnostics->diagnostic-messages diags-by-uri options db))))
-       :diagnostics diags-by-uri}
-      {:result-code 0 :message-fn (constantly "No diagnostics found!")})))
+(defn ^:private filter-diff
+  [diff-out diags-by-uri]
+  (let [diags-by-uri-keys (keys diags-by-uri)
+        chunks (diff/->chunks diff-out)
+        chunks-with-additions (filter #(-> % :added-line-numbers seq) chunks)
+        additions-by-file (reduce (fn [acc {:keys [file] :as hunk}]
+                                    (update acc file (fnil conj []) hunk))
+                                  {}
+                                  chunks-with-additions)
+        additions-by-uri (reduce (fn [acc [file hunks]]
+                                   (if-let [uri (some #(when (string/ends-with? % file) %)
+                                                      diags-by-uri-keys)]
+                                     (assoc acc uri hunks)
+                                     acc))
+                                 {}
+                                 additions-by-file)
+        diags-by-uri-added (reduce (fn [acc [uri hunks]]
+                                     (assoc acc
+                                            uri
+                                            (filter (fn [{{{line :line} :start} :range}]
+                                                      (some (fn [{:keys [added-line-numbers]}]
+                                                              (contains? added-line-numbers line))
+                                                            hunks))
+                                                    (get diags-by-uri uri))))
+                                   {}
+                                   additions-by-uri)]
+    diags-by-uri-added))
+
+(defn ^:private diagnostics* [{{:keys [format]} :output rev-range :diff :as options} {:keys [db*] :as components}]
+  (let [{diff-exit :exit diff-out :out diff-err :err}
+        (if rev-range
+          (sh/sh "git" "diff" rev-range "--" "*.clj")
```

Comment:
not sure I got this here, wy we are git diffing before analyze the project? shouldn't we do this later?